server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -1229,8 +1229,8 @@ public VolumeVO resizeVolume(ResizeVolumeCmd cmd) throws ResourceAllocationExcep
 
                 validateIops(newMinIops, newMaxIops, volume.getPoolType());
             } else {
-                newMinIops = newDiskOffering.getMinIops();
-                newMaxIops = newDiskOffering.getMaxIops();
+                newMinIops = newDiskOffering.getMinIops() != null ? newDiskOffering.getMinIops() : newDiskOffering.getIopsReadRate();
+                newMaxIops = newDiskOffering.getMaxIops() != null ? newDiskOffering.getMaxIops() : newDiskOffering.getIopsWriteRate();
```

Comment:
why get read/write rate if the min/max are null?

Reply:
AFAIK, VMware only allows IOPS limitation, if the offering has any kind of limitation, we should set this limitation.